taigit/src/libraries/Taiga.ts

```diff
@@ -78,4 +78,118 @@ task_history(taskId : number) : Promise<Object> {
     }
 
     return output;
-}
\ No newline at end of file
+}
+
+/**
+ * @summary This call assess a task based on task_assessment_state_trans based on task Id
+ * @param taskId the ID for the task to assess task
+ * @returns  whether the task's state transition is valid,task status transition, date based on task Id
+ * * {
+ *      state_trans_invalid : Boolean       // true, the task state transition is not valid; false, the task is valid.
+ *      date : number,         // Date and time of history entry in milliseconds since epoch
+ *      user : object,         // Taiga User Object
+ *      status_trans:entry.String[],//Status transition array
+ *
+ * }
+ */
+export async function
+task_assessment_state_trans(taskId : number) : Promise<Object> {
+let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
+    //test case :https://api.taiga.io/api/v1/history/task/2577741
+let output : Array<Object> = [];
+for(let entry of data) {
+    let new_entry = {
+        state_trans_invalid : false,
+        date : new Date(entry.created_at).getTime(),
+        user : entry.user,
+        status_trans:entry.values_diff.status
+    }
+
+    //Only three status transistion is valid
+    //which is  ["New", "In progress"] ["In progress", "Ready for test"]  ["Ready for test", "Closed"])
+    if((new_entry.status_trans[0] != "New"&&new_entry.status_trans[1] != "In progress")
+        &&(new_entry.status_trans[0] != "In progress"&&new_entry.status_trans[1] != "Ready for test")
+        &&(new_entry.status_trans[0] != "Ready for test"&&new_entry.status_trans[1] != "Closed"))
+        new_entry.state_trans_invalid = true;
+    output.push(new_entry);
+}
+
+return output;
+}
+
+
+/**
+ * @summary This call return a task current end status and estimated finished percentage based on task Id
+ * @param taskId the ID for the task to assess task
+ * @returns  whether the task is abnormal,task status transition, date based on task Id
+ * * {
+ *      finished : Boolean           // true, the task is finished.
+ *      fin_per : number            // finished percentage % default [0%,10%,60%,100%]
+ *      end_status : string,       // current task status  ["New", "In progress","Ready for test", "Closed"]
+ *      num_stat : number           // current task status [0 1 2 3] corresponding to string in end_status
+ *
+ * }
+ */
+export async function
+task_assessment_endstate(taskId : number) : Promise<Object> {
+
+    let data = (await axios.get(`https://api.taiga.io/api/v1/history/task/${taskId}`)).data;
```

Comment:
we already have a task history function. why repeat the call?

Reply:
I do not know the other way to get the information I required.

Reply:
I made a function called `task_history()` that does exactly this call.

Reply:
However, I get the data from the task history, and provide further processing with the data.